src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -724,7 +724,20 @@ void BaseCouplingScheme::advanceTXTWriters()
 
 bool BaseCouplingScheme::reachedEndOfTimeWindow()
 {
-  return math::equals(getNextTimeStepMaxSize(), 0.0, _eps) || not hasTimeWindowSize();
+  if (math::equals(getNextTimeStepMaxSize(), 0.0, _minTimeStepSize) && getNextTimeStepMaxSize() > math::NUMERICAL_ZERO_DIFFERENCE && hasTimeWindowSize()) {
+    for (auto &data : _allData | boost::adaptors::map_values) {
+      auto   timeStepStorage = data->timeStepsStorage();
+      auto   values          = timeStepStorage.getTimesAndValues();
+      auto   ts              = values.first;
+      double endOfTimeWindow = getTime() - _computedTimeWindowPart + _timeWindowSize;
```

Comment:
```suggestion
      double endOfTimeWindow = getEndOfTimeWindow();
```

Please implement a dedicated function here. This is hard to understand otherwise (and you are also duplicating this statement below). Did something similar in #1822, but this is not merged yet. See `BaseCouplingScheme::getWindowEndTime()`.